packages/styles/src/components/button.scss

```diff
@@ -134,22 +134,48 @@
 }
 
 // Animated
+.btn-animated,
+.btn-animated-left {
+  &:not(.btn-link, .btn-tertiary) {
+    &::after {
+      content: '';
+      position: absolute;
+      width: 0.625em;
+      height: 0.625em;
+      transform-origin: center center;
+      transition:
+        opacity 250ms,
+        transform 250ms;
+      opacity: 0;
+    }
+
+    > span {
+      transition: transform 250ms;
+    }
+
+    @media (prefers-reduced-motion: no-preference) {
+      @include utilities.not-disabled-focus-hover() {
+        &::after {
+          transform: rotateZ(315deg) translate(0, 0);
+          opacity: 1;
+        }
+      }
+
+      > span {
+        // Initially transform to place text in the right rendering context for a smooth animation
+        transform: translateX(0);
+      }
+    }
+  }
+}
+
 .btn-animated:not(.btn-link, .btn-tertiary) {
   &::after {
-    content: '';
-    position: absolute;
     right: button.$btn-padding-x-md - button.$btn-animation-distance-md;
-    width: 0.625em;
-    height: 0.625em;
     transform: rotateZ(315deg)
       translate(button.$btn-border-width * -1, button.$btn-border-width * -1);
-    transform-origin: center center;
-    transition:
-      opacity 250ms,
-      transform 250ms;
     border-right: button.$btn-border-width solid currentColor;
```

Comment:
This PR would be a good opportunity to use proper chevron icons with `@include icons-mx.icon(2110);` and `@include icons-mx.icon(2111);`. This will simplify the code and help match the design even more, what do you think? 
If you prefer you can simply open a new ticket so that we do it later.